lib/gl_rubocop/gl_cops/no_stubbing_perform_async.rb

```diff
@@ -0,0 +1,59 @@
+module GLRubocop
+  module GLCops
+    # This cop ensures that you don't stub perform_async. Stubbing prevents seeing errors from
+    # invalid argumens - because sidekiq arguments have to be valid JSON data types:
+    # github.com/sidekiq/sidekiq/wiki/Best-Practices#1-make-your-job-parameters-small-and-simple
+    #
+    # Good:
+    #   expect(ExampleWorker).to have_enqueued_sidekiq_job
+
+    #
+    # Bad:
+    #   allow(ExampleWorker).to receive(:perform_async)
+    #   expect(ExampleWorker).to receive(:perform_async)
+    #   expect(SomeWorker).not_to have_received(:perform_in)
+    class NoStubbingPerformAsync < RuboCop::Cop::Cop
+      MSG = "Don't stub perform async. Use the rspec-sidekick matchers instead: " \
+            'expect(JobClass).to have_enqueued_sidekiq_job'.freeze
+
+      # This pattern captures expectations for perform_async and perform_in
+      # expect(SomeWorker).not_to have_received(:perform_async)
+      # expect(SomeWorker).to have_received(:perform_async)
+      # expect(SomeWorker).to have_received(:perform_in)
+      def_node_matcher :perform_method_expectation?, <<~PATTERN
+        (send#{' '}
+          (send nil? :expect $_)#{' '}
+          {:not_to :to_not :to}#{' '}
+          (send nil? :have_received (sym {:perform_async :perform_in}) ...))
```

Comment:
I see an example of us also stubbing just the `perform` method [here](https://github.com/givelively/charity-api/blob/b47dfc3d72d2a42f821da920a4a66646bc9e8544/spec/support/capybara.rb#L21). Should that be included as a violation of this rule as well?